File: FashionShop.Backend/FashionShop.Core/Entities/Voucher.cs
using FashionShop.Core.Common;
using FashionShop.Core.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FashionShop.Core.Entities
{
    [Table("Vouchers")]
    public class Voucher : BaseEntity
    {
        [Required]
        [MaxLength(200)]
        public string Name { get; set; }

        [Required]
        [MaxLength(50)]
        [Column(TypeName = "varchar(50)")]
        public string Code { get; set; }

        [Required]
        [MaxLength(500)]
        public string Description { get; set; }

        [Required]
        public DiscountType DiscountType { get; set; } = DiscountType.FixedAmount;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal DiscountAmount { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? MaxDiscountAmount { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal MinOrderValue { get; set; }

        [Required]
        public int Quantity { get; set; }

        [Required]
        public int UsedCount { get; set; } = 0;

        [Required]
        public int MaxUsagePerUser { get; set; } = 1;

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        public bool IsActive { get; set; } = true;

        public virtual ICollection<Order> Orders { get; set; } = new List<Order>();

        public virtual ICollection<VoucherUsage> VoucherUsages { get; set; } = new List<VoucherUsage>();
    }
}

File: FashionShop.Web/src/features/admin/vouchers/types/voucher.ts
export interface Voucher {
    id: string;
    name: string;
    code: string;
    description: string;
    discountType: DiscountType;
    discountAmount: number;
    maxDiscountAmount?: number;
    minOrderValue: number;
    quantity: number;
    usedCount: number;
    maxUsagePerUser: number;
    startDate: Date;
    endDate: Date;
    isActive: boolean;
    createdDate: Date;
    updatedDate: Date;
}

export const DiscountType = {
    FixedAmount: "FixedAmount",
    Percentage: "Percentage",
} as const;

export type DiscountType = typeof DiscountType[keyof typeof DiscountType];